lib/src/command/add.dart

```diff
@@ -391,11 +392,18 @@ class AddCommand extends PubCommand {
 
     /// Remove the package from dev_dependencies if we are adding it to
     /// dependencies. Refer to [_addPackageToPubspec] for additional discussion.
-    if (!isDevelopment &&
-        yamlEditor.parseAt(['dev_dependencies', package.name],
-                orElse: () => null) !=
-            null) {
-      yamlEditor.remove(['dev_dependencies', package.name]);
+    if (!isDevelopment) {
+      final devDependenciesNode =
+          yamlEditor.parseAt(['dev_dependencies'], orElse: () => null);
+
+      if (devDependenciesNode is YamlMap &&
```

Comment:
I don't think you need to import `package:yaml/yaml.dart` and use `devDependenciesNode is YamlMap`, you can just do:
```suggestion
      if (devDependenciesNode is Map &&
```

I think? :)

Reply:
ah the `YamlMap` is there so that I can both do the `containsKey` check and `length` check in the subsequent lines! If I swapped it to just `Map`, the Dart linter(?) doesn't seem to resolve that the `devDependenciesNode` has the `containsKey` method and `length` property automatically. Would substituting the current code with:

```dart
      if (devDependenciesNode is Map &&
          (devDependenciesNode as Map).containsKey(package.name)) {
        if ((devDependenciesNode as Map).length == 1) {
```

be preferred?

Reply:
interesting... I think we YamlMap is fine :D